hikey960.mk

```diff
@@ -0,0 +1,444 @@
+		ARCH=arm64 \
+		LOCALVERSION= \
+		gen_init_cpio
+
+LINUX_COMMON_FLAGS += ARCH=arm64 Image modules hisilicon/hi3660-hikey960.dtb
+
+.PHONY: linux
+linux: linux-common
+
+.PHONY: linux-defconfig-clean
+linux-defconfig-clean: linux-defconfig-clean-common
+
+LINUX_CLEAN_COMMON_FLAGS += ARCH=arm64
+
+.PHONY: linux-clean
+linux-clean: linux-clean-common
+
+LINUX_CLEANER_COMMON_FLAGS += ARCH=arm64
+
+.PHONY: linux-cleaner
+linux-cleaner: linux-cleaner-common
+
+################################################################################
+# OP-TEE
+################################################################################
+OPTEE_OS_COMMON_FLAGS += PLATFORM=hikey-hikey960 \
+			CFG_CONSOLE_UART=$(CFG_CONSOLE_UART) \
+			CFG_SECURE_DATA_PATH=n
+OPTEE_OS_CLEAN_COMMON_FLAGS += PLATFORM=hikey-hikey960
+
+.PHONY: optee-os
+optee-os: optee-os-common
+
+.PHONY: optee-os-clean
+optee-os-clean: optee-os-clean-common
+
+.PHONY: optee-client
+optee-client: optee-client-common
+
+.PHONY: optee-client-clean
+optee-client-clean: optee-client-clean-common
+
+################################################################################
+# xtest / optee_test
+################################################################################
+.PHONY: xtest
+xtest: xtest-common
+
+# FIXME:
+# "make clean" in xtest: fails if optee_os has been cleaned previously
+.PHONY: xtest-clean
+xtest-clean: xtest-clean-common
+	rm -rf $(OPTEE_TEST_OUT_PATH)
+
+.PHONY: xtest-patch
+xtest-patch: xtest-patch-common
+
+################################################################################
+# hello_world
+################################################################################
+.PHONY: helloworld
+helloworld: helloworld-common
+
+.PHONY: helloworld-clean
+helloworld-clean: helloworld-clean-common
+
+################################################################################
+# strace
+################################################################################
+.PHONY: strace
+strace:
+	cd $(STRACE_PATH); \
+	./bootstrap; \
+	set -e; \
+	./configure --host=$(MULTIARCH) CC="$(CCACHE)$(AARCH$(COMPILE_NS_USER)_CROSS_COMPILE)gcc" LD=$(AARCH$(COMPILE_NS_USER)_CROSS_COMPILE)ld; \
+	CC="$(CCACHE)$(AARCH$(COMPILE_NS_USER)_CROSS_COMPILE)gcc" LD=$(AARCH$(COMPILE_NS_USER)_CROSS_COMPILE)ld $(MAKE) -C $(STRACE_PATH)
+
+.PHONY: strace-clean
+strace-clean:
+	if [ -e $(STRACE_PATH)/Makefile ]; then $(MAKE) -C $(STRACE_PATH) clean; fi
+
+.PHONY: strace-cleaner
+strace-cleaner: strace-clean
+	rm -f $(STRACE_PATH)/Makefile $(STRACE_PATH)/configure
+
+################################################################################
+# Root FS
+################################################################################
+# Read stdin, expand ${VAR} environment variables, output to stdout
+# http://superuser.com/a/302847
+define expand-env-var
+awk '{while(match($$0,"[$$]{[^}]*}")) {var=substr($$0,RSTART+2,RLENGTH -3);gsub("[$$]{"var"}",ENVIRON[var])}}1'
+endef
```

Comment:
Duplicated from `hikey.mk`. Should be moved to a common file, such as `common.mk`.